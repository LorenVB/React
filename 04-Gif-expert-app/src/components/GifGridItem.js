import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ( {title, url} ) => {
  
    return (
    <div>
       <img className="gif-img animate__animated animate__bounce animate__fadeIn" src={url} alt={ title } />
       <p> {title} </p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}


