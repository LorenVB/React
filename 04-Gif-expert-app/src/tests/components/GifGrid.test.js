import React from 'react';
import {shallow} from 'enzyme'
import { GifGrid } from "../../components/GifGrid"
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Testing GifGrid component', () => { 
    const category = 'Goku';

    test('should render properly', () => { 

        useFetchGifs.mockReturnValue({
            data: [],
            loading:true
        })
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();
     })

     test('should return items when loading is false by useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/anything',
            title: 'Anything'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading:false
        })
         
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
         
     })

     
 })