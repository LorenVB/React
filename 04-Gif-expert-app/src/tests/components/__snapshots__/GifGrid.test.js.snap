// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing GifGrid component should render properly 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <p
    className="animate__flash"
  >
    Loading
  </p>
  <div
    className="gif-container"
  />
</Fragment>
`;

exports[`Testing GifGrid component should return items when loading is false by useFetchGifs 1`] = `
<Fragment>
  <h3>
    Goku
  </h3>
  <div
    className="gif-container"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Anything"
      url="https://localhost/anything"
    />
  </div>
</Fragment>
`;
